In [1]:
### Q4.8
install.packages('Stat2Data', repos='http://cran.us.r-project.org')
library(Stat2Data)
data(CountyHealth)
str(CountyHealth)

Installing package into ‘/home/jupyter/R/x86_64-pc-linux-gnu-library/4.2’
(as ‘lib’ is unspecified)



'data.frame':	53 obs. of  4 variables:
 $ County   : Factor w/ 53 levels "Bay, FL                             ",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ MDs      : int  351 95 260 2797 769 42 13 20 2981 83 ...
 $ Hospitals: int  3 2 2 11 5 2 2 2 7 3 ...
 $ Beds     : int  605 134 567 1435 976 245 33 65 1462 100 ...


In [2]:
# part a
train = CountyHealth[1:35,]
holdout = CountyHealth[36:53,]

fit_model = lm(sqrt(MDs)~Hospitals, data=train)
summary(fit_model)


Call:
lm(formula = sqrt(MDs) ~ Hospitals, data = train)

Residuals:
    Min      1Q  Median      3Q     Max 
-18.582  -6.362  -2.918   8.277  23.170 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -3.1695     2.6915  -1.178    0.247    
Hospitals     6.7853     0.5284  12.841 2.19e-14 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 9.627 on 33 degrees of freedom
Multiple R-squared:  0.8332,	Adjusted R-squared:  0.8282 
F-statistic: 164.9 on 1 and 33 DF,  p-value: 2.194e-14


Root-MDS-hat = -3.1695 + 6.7853Hospitals

In [3]:
# part b
hold_df = data.frame(Hospitals = holdout$Hospitals)
pridiction = predict(fit_model, newdata = hold_df)
pridiction_square = pridiction^2
cor(pridiction_square, holdout$MDs)

[1] 0.9778562

part c \
Shrinkage = 0.8332 - 0.9778^2 = -0.1229 \
This model is great for predicting MDs because holdout data produces correlation great